cnudie/iter.py

```diff
@@ -140,17 +162,43 @@ def inner_iter(
                 name=cref.componentName,
                 version=cref.version,
             )
+
             if ocm_repo:
                 referenced_component_descriptor = lookup(cref_id, ocm_repo)
             else:
                 referenced_component_descriptor = lookup(cref_id)
-            referenced_component = referenced_component_descriptor.component
 
             yield from inner_iter(
-                component=referenced_component,
+                component=referenced_component_descriptor.component,
+                lookup=lookup,
+                recursion_depth=recursion_depth,
+                path=path,
+                reftype=reftype,
+            )
+
+        if not (extra_crefs_label := component.find_label(
+            name=dso.labels.ExtraComponentReferencesLabel.name,
+        )):
+            return
+
+        extra_crefs_label: dso.labels.ExtraComponentReferencesLabel = dso.labels.deserialise_label(
+            label=extra_crefs_label,
+        )
+
+        for extra_cref in extra_crefs_label.value:
+            extra_cref_id = extra_cref.component_reference
+
+            if ocm_repo:
+                referenced_component_descriptor = lookup(extra_cref_id, ocm_repo)
+            else:
+                referenced_component_descriptor = lookup(extra_cref_id)
+
+            yield from inner_iter(
+                component=referenced_component_descriptor.component,
                 lookup=lookup,
                 recursion_depth=recursion_depth,
                 path=path,
+                reftype=NodeReferenceType.EXTRA_COMPONENT_REFS_LABEL,
```

Comment:
IMHO, we should only mark the exact component that declared reference via label (sub-components from such a component should again be marked as regular crefs (if they are regular crefs, of course)

Reply:
I was not sure about this either, so I'm with you to define it like that :-)